WooCommerce/Classes/ViewModels/Order Details/Shipping Labels/ReprintShippingLabelViewController.swift

```diff
@@ -0,0 +1,257 @@
+            // TODO-2169: Present printing instructions modal
+            break
+        default:
+            return
+        }
+    }
+}
+
+// MARK: Cell configuration
+private extension ReprintShippingLabelViewController {
+    func configure(_ cell: UITableViewCell, for row: Row) {
+        switch cell {
+        case let cell as BasicTableViewCell where row == .headerText:
+            configureHeaderText(cell: cell)
+        case let cell as TopLeftImageTableViewCell where row == .infoText:
+            configureInfoText(cell: cell)
+        case let cell as SettingTitleAndValueTableViewCell where row == .paperSize:
+            configurePaperSize(cell: cell)
+        case let cell as SpacerTableViewCell where row == .spacer:
+            configureSpacer(cell: cell)
+        case let cell as TopLeftImageTableViewCell where row == .paperSizeOptions:
+            configurePaperSizeOptions(cell: cell)
+        case let cell as TopLeftImageTableViewCell where row == .printingInstructions:
+            configurePrintingInstructions(cell: cell)
+        default:
+            break
+        }
+    }
+
+    func configureHeaderText(cell: BasicTableViewCell) {
+        cell.textLabel?.text = Localization.headerText
+        cell.textLabel?.numberOfLines = 0
+        cell.textLabel?.applyBodyStyle()
+        cell.hideSeparator()
+    }
+
+    func configureInfoText(cell: TopLeftImageTableViewCell) {
+        cell.imageView?.image = .infoOutlineImage
+        cell.imageView?.tintColor = .systemColor(.secondaryLabel)
+        cell.textLabel?.text = Localization.infoText
+        cell.textLabel?.applyBodyStyle()
+        cell.textLabel?.textColor = .systemColor(.secondaryLabel)
+        cell.hideSeparator()
+    }
+
+    func configurePaperSize(cell: SettingTitleAndValueTableViewCell) {
+        cell.updateUI(title: Localization.paperSizeSelectorTitle, value: selectedPaperSize?.description)
+        cell.accessoryType = .disclosureIndicator
+    }
+
+    func configureSpacer(cell: SpacerTableViewCell) {
+        cell.height = Constants.verticalSpacingBetweenPaperSizeSelectorAndInfoLinks
+    }
+
+    func configurePaperSizeOptions(cell: TopLeftImageTableViewCell) {
+        cell.imageView?.image = .pagesFootnoteImage
+        cell.textLabel?.text = Localization.paperSizeOptionsButtonTitle
+        configureCommonStylesForInfoLinkCell(cell)
+    }
+
+    func configurePrintingInstructions(cell: TopLeftImageTableViewCell) {
+        cell.imageView?.image = .infoOutlineFootnoteImage
+        cell.textLabel?.text = Localization.printingInstructionsButtonTitle
+        configureCommonStylesForInfoLinkCell(cell)
+    }
+
+    func configureCommonStylesForInfoLinkCell(_ cell: TopLeftImageTableViewCell) {
+        cell.imageView?.tintColor = .systemColor(.secondaryLabel)
+        cell.textLabel?.applyFootnoteStyle()
+        cell.textLabel?.textColor = .systemColor(.secondaryLabel)
+        cell.hideSeparator()
+        cell.selectionStyle = .default
+    }
```

Comment:
ℹ️  I think we should consider this as a legitimate configuration style for `TopLeftImageTableViewCell`. Something like:

```swift
cell.configure(style: .secondaryOrWhatever)
```

This is similar to what we did with [`HeadlineLabelTableViewCell`](https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/ViewRelated/ReusableViews/HeadlineLabelTableViewCell.swift):

https://github.com/woocommerce/woocommerce-ios/blob/b8e1d3b3b4b06b8a13b988ef98c64c0a37481f65/WooCommerce/Classes/ViewRelated/ReusableViews/HeadlineLabelTableViewCell.swift#L53-L66

The theory is that:

- We have standard “styles” to use all throughout the app.
- ViewControllers would only be using styles and not defining their own standards.
- Probably makes migration to SwiftUI easier. Probably. 

Something to think about in the future. 🙂

Reply:
Great suggestion! It is nice to have a set of predefined styles. So far, I just see the consistent usage of font size for `TopLeftImageTableViewCell ` while the image/text colors differ in use cases. Updated in: https://github.com/woocommerce/woocommerce-ios/pull/3342/commits/830f6aede4fbbd41467f01c0b8bbe579831d0871